src/components/reviewMatches/ReviewLink.tsx

```diff
@@ -194,90 +179,117 @@ const ReviewLink = () => {
     return <ApiErrorMessage error={error} />
   }
 
-  if (!data) {
+  if (!matchDetails) {
     return <NotFound />
   }
 
   const handleOpenLinkedRecordDialog = (uid: string) => {
-    const tableDataTemp: AnyRecord[] = data.filter(d => {
-      if (d.uid === uid || d.type === 'Golden') {
+    const tableDataTemp = candidateGoldenRecords?.filter(d => {
+      if (d.uid === uid) {
         return d
       }
     })
 
-    setTableData(tableDataTemp)
+    if (tableDataTemp && patientRecord)
```

Comment:
tableDataTemp is an array, even if empty it's evaluated to "true"
We could directly set the state, without having the condition.

Reply:
We can't because `tableDataTemp` is of type `AnyRecord[] | undefined`

Reply:
Let's change the type to AnyRecord[]